### PR TITLE
DEPR: make Categorical.ravel() return Categorical

### PR DESCRIPTION
**Pull request by jbrockmendel**
Discussed in #27142.

- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by TomAugspurger — APPROVED** [review]

**Comment by jreback** [comment]
conflict and failing

```
pandas/tests/extension/test_categorical.py(100,): error :  Found unwanted pattern:         with pytest.warns(FutureWarning, match="will return a Categorical")
```
otherwise lgtm.

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -830,6 +830,7 @@ Other deprecations
 - :meth:`Index.item` and :meth:`Series.item` is deprecated. (:issue:`18262`)
 - The default value ``ordered=None`` in :class:`~pandas.api.types.CategoricalDtype` has been deprecated in favor of ``ordered=False``. When converting between categorical types ``ordered=True`` must be explicitly passed in order to be preserved. (:issue:`26336`)
 - :meth:`Index.contains` is deprecated. Use ``key in index`` (``__contains__``) instead (:issue:`17753`).
+- :meth:`DataFrame.get_dtype_counts` is deprecated. (:issue:`18262`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: I think this is a dupe

File: pandas/tests/extension/base/base.py [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 
 class BaseExtensionTests:
+    # Whether the EA being tested supports __setitem__
+    _supports_setitem = True
```
Comment (jreback) [inline pandas/tests/extension/base/base.py]: I guess this is ok, but in reality you should have a class which does this (future PR)

**Comment by jreback** [comment]
I am a little concerned by the change in testing pattern this introduces. Could we remove this for now.

**Comment by jbrockmendel** [comment]
testing pattern reverted

**Comment by jreback** [comment]
lgtm. merge on green.

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**